book/src/exec.md

```diff
@@ -0,0 +1,174 @@
+
+![exec_high.png](img/exec_high.png)
+
+## Step 1: Create an intent
+
+Intents specify ephemeral interests of users. The step of intent creation consists of two parts:
+1. **Specify an intent**. `IntentVP` encodes the user interests in the [VP](./validity-predicates.md) form. As in a valid transaction all VPs must evaluate to `True`, the intent is guaranteed to be satisfied.
+2. **Create the initial *partial transaction***.
+
+### Partial transaction
+
+Two of the main requirements for an Anoma transaction to be valid are: 
+1. VPs of all involved in the transaction parties must be satisfied.
+2. The transaction must be **balanced** , i.e the value spent in the transaction minus the value output equals to some predefined value `v`, the **balancing value**. The balancing value is transparent (visible to observers) and can be involved in interactions with the transparent system.
+
+We call a **partial transaction** a state transition where the first requirement is satisfied (all VPs evaluate to True), but the second one isn't (the state transition is unbalanced). Such a state transition isn't a valid transaction and cannot be published on the blockchain, 
+but can be combined with other partial transactions in order to build a valid transaction. 
+
+![img.png](img/exec_partial_tx.png)
+
+Partial transactions are immutable objects, and once they are created, they cannot be changed. 
+In principle, any party can create a partial transaction, if they can prove that the VPs of the parties involved in the partial transaction are satisfied. 
+
+**Initial partial transactions** are the partial transactions that users create at the beginning, spending the notes that will be given to the counterparty once the intents are matched.
+
+**Note**: For simplicity we assume that intents imply both sending notes and receiving notes.
+
+As a part of the initial partial transaction, users also output **intent token notes**. 
+### Intent tokens
+
+**Intent tokens** are a mechanism that makes sure that the partial transaction is *unbalanced* until the corresponding intent is satisfied, making it impossible to publish the transaction.
+
+Intent token notes are **dummy** notes - meaning that unlike "normal" notes, the merkle path isn't checked for them (but they can have arbitrary value and stored in the CMtree, just like "normal"). 
+
+**Note**: being a dummy note and having zero value are two independent concepts in Taiga. Zero value notes aren't necessarily dummy, dummy notes can have non-zero value.
+
+How intent tokens are used:
+1. Spending notes in the initial partial transaction, the user additionally outputs intent token note of value [1].
+This note will only be spent if the user's intent (intentVP) is satisfied.
+2. Once all of the parties involved in a state transition spent their intent token notes (meaning that the intents of all of them are satisfied), the transaction is balanced and can be finalized.
```

Comment:
So iiuc, each user creates a (intent token) note with what they want to send and a boolean (`[1]~true`). Would it be possible to do the same thing without the intent token note?

Reply:
maybe it is simply an abstraction for the "message passing" explained in the next paragraph?

Reply:
In a simple example, each user spends a note they want to give away and creates an intent token note of value [1]. Intents tokens are here to make sure the transaction is unbalanced if any of the parties didn't get what they wanted. Once there is another ptx where the user receives whatever they wanted, the user spends this intent token note, balancing the tx

Reply:
I see. I was wondering if it is not possible to achieve this without this `[1]`, in order to simplify a bit, but maybe it is not that simple :-D